File: src/sql/postgres/protocol/CopyInResponse.zig
const CopyInResponse = @This();

pub fn decodeInternal(this: *@This(), comptime Container: type, reader: NewReader(Container)) !void {
    _ = reader;
    _ = this;
    bun.Output.panic("TODO: not implemented {s}", .{bun.meta.typeBaseName(@typeName(@This()))});
}

pub const decode = DecoderWrap(CopyInResponse, decodeInternal).decode;

const bun = @import("bun");
const DecoderWrap = @import("./DecoderWrap.zig").DecoderWrap;
const NewReader = @import("./NewReader.zig").NewReader;

File: src/sql/postgres/protocol/CopyInResponse.rs
use super::new_reader::NewReader;

pub struct CopyInResponse;

impl CopyInResponse {
    pub fn decode_internal<Container: super::new_reader::ReaderContext>(
        &mut self,
        reader: NewReader<Container>,
    ) -> Result<(), crate::Error> {
        drop(reader);
        let _ = self;
        bun_core::output::panic(format_args!("TODO: not implemented {}", "CopyInResponse"));
    }

    pub fn decode<Container: super::new_reader::ReaderContext>(
        &mut self,
        context: Container,
    ) -> Result<(), crate::Error> {
        self.decode_internal(NewReader { wrapped: context })
    }
}
